packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/container.tsx

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { RouteComponentProps, useLocation } from '@reach/router';
+import { currentAccount } from '../../../lib/cache';
+import { useFinishOAuthFlowHandler } from '../../../lib/oauth/hooks';
+import {
+  Integration,
+  isOAuthIntegration,
+  useAuthClient,
+} from '../../../models';
+import AppLayout from '../../../components/AppLayout';
+import CardHeader from '../../../components/CardHeader';
+import ConfirmSignupCode from '.';
+import { hardNavigateToContentServer } from 'fxa-react/lib/utils';
+import LoadingSpinner from 'fxa-react/components/LoadingSpinner';
+import { LocationState } from './interfaces';
+
+const SignupConfirmCodeContainer = ({
+  integration,
+}: {
+  integration: Integration;
+} & RouteComponentProps) => {
+  const authClient = useAuthClient();
+  const location = useLocation() as ReturnType<typeof useLocation> & {
+    state: LocationState;
+  };
+  const {
+    selectedNewsletterSlugs: newsletterSlugs,
+    keyFetchToken,
+    unwrapBKey,
+  } = location.state || {};
+
+  const { finishOAuthFlowHandler, oAuthDataError } = useFinishOAuthFlowHandler(
+    authClient,
+    integration
+  );
+
+  // TODO: UX for this, FXA-8106
+  if (oAuthDataError) {
+    return (
+      <AppLayout>
+        <CardHeader
+          headingText="Unexpected error"
+          headingTextFtlId="auth-error-999"
+        />
+      </AppLayout>
+    );
+  }
+
+  /* Users who reach this page should have account data set in localStorage.
+   * Account data is persisted local storage after creating an (unverified) account
+   * and after sign in. Users may also have localStorage set by the browser if
+   * they are logged in.
+
+   * The session token from local storage is required to verify the signup code.
+   * If this page is reached without an account in localStorage (e.g., directly from
+   * URL), redirect to `/`.
+
+   * TOOD: when we pull the account.verifySession call into the container component,
+   * ensure we're only reading from localStorage once. `sessionToken()` also reads from
+   * localStorage. */
+  const storedLocalAccount = currentAccount();
+  const { email, sessionToken, uid } = storedLocalAccount;
+
+  // Users in this state should never reach this as they should see the Backbone version
```

Comment:
Do we want to emit a metric or fabricated sentry error to make sure / monitor this state isn't happneing...

Reply:
We could if you think it's worth doing, that might be faster than a functional test checking to see if it's React or Backbone. We can also call this out to QA.

Reply:
I don't really have a preference. Definitely not a blocker on this PR.